rust/ql/lib/codeql/rust/internal/TypeInference.qll

```diff
@@ -829,7 +767,7 @@ private Type inferCallExprBaseType(AstNode n, TypePath path) {
     n = a.getNodeAt(apos) and
     result = CallExprBaseMatching::inferAccessType(a, apos, path0)
   |
-    if apos.isSelf()
+    if apos.isSelf(_)
```

Comment:
Shouldn't this be `true`?

Reply:
I thought so as well, but for some reason it didn't work for operator calls. Extra `&` types where inferred around operator calls.

Reply:
I think it would make sense to investigate this further in follow up work.